backend/auth.ts

```diff
@@ -80,7 +114,11 @@ export async function requireLogin(
   res: NextApiResponse,
 ): Promise<{ user: LuciaUser; session: Session }> {
   const userDetails = await validateRequest(req, res);
-  if (!userDetails.session || !userDetails.user) {
+  if (
+    !userDetails.session ||
+    !userDetails.user ||
+    !userDetails.user.isLoggedIn
+  ) {
```

Comment:
Tein sen muutoksen, että käyttäjän `isLoggedIn`-arvo pitää olla `true`, jotta pääsee "sovellukseen sisään"

Eli tässä tarkistetaan puuttuuko `session`- tai `user`-objecti tai onko käyttäjän `isLoggedIn` `false`

Reply:
Jes juuri näin ajattelin!

Reply:
Sori ajatusvirhe mulla. Tuo `isLoggedIn` pitäisi tallentaa sessioon, ei käyttäjään. On mahdollista että käyttäjällä on useita sessioita yhtä aikaa olemassa. Esim. jos siirtyy kirjaston koneelta omalle koneelle ja kirjautuu omalla koneella merkitään käyttäjälle `isLoggedIn = true` jolloin kirjaston koneelle jäänyt sessio olisikin yht äkkiä kirjautuneena sisään.

Reply:
Olin lukenut väärin sun idean Discordista, joten meikän moka! Muokkaan tuon samointein :+1: